Document Micronaut Context dependency requirement for immutable configuration properties, scheduling, and context events. (#10095)
micronaut-projects/micronaut-core: src/main/docs/guide/config/immutableConfig.adoc

> Since 1.3, Micronaut framework supports the definition of immutable configuration. To enable mutable configuration, add the following dependency on Micronaut Context:

Reviewer: ```suggestion
Since 1.3, Micronaut framework supports the definition of immutable configuration. Immutable configuration with an interface requires the Micronaut Context dependency.
```

> dependency::micronaut-context[]

Reviewer: ```suggestion
dependency::micronaut-context[]

`micronaut-context` is a transitive dependency of `micronaut-http`. If you use a Micronaut HTTP runtime, your project already includes the `Micronaut-context` dependency. 
```
